### PR TITLE
task3 completed

### PR DESCRIPTION
**Pull request by FameTii**

### CODE REVIEW

File: API/app.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+const express = require("express");
+const app = express();
+app.use(express.json());
```
Comment (Augs0) [inline API/app.js]: this shouldn't be here until we start posting

File: API/app.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+const { getCategories } = require("./controller");
+
+app.get("/api/categories", getCategories);
+
+app.use((err, req, res, next) => {
+  if (err.msg === "invalid sort_by option") {
```
Comment (Augs0) [inline API/app.js]: Where is this coming from? We are not handling sort queries yet

File: API/app.test.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
+const request = require("supertest");
+const { seed } = require("../db/seeds/seed");
```
Comment (Augs0) [inline API/app.test.js]: Unless your seed file is exported in an object, you won't need to import it in an object

File: API/app.test.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+describe('GET /api/categories', () => {
+    it('respond with 200 and return an array of category objects and includes properties of slug and description', () => {
+        return request(app)
+            .get("/api/categories")
+            .expect(200)
+            .then(({body}) => {
+                expect(typeof body).toBe('object')
+                const {categories} = body
+                expect(body.categories).toBeInstanceOf(Array);
+                categories.forEach((category) => {
```
Comment (Augs0) [inline API/app.test.js:24]: Before iterating over an array, we should check the length of it. There is a jest matcher for this :)

File: API/models.js [hunk 5]
```diff
@@ -0,0 +1,9 @@
+const db = require('../db/connection')
+
+exports.fetchCategories = () => {
+    let queryStr = `SELECT slug, description FROM categories`
+    return db.query(queryStr).then((result) => {
+        // console.log(result.rows);
```
Comment (Augs0) [inline API/models.js]: Please remove console logs before pushing code

File: db/seeds/seed.js [hunk 6]
```diff
@@ -125,4 +125,4 @@ const seed = ({ categoryData, commentData, reviewData, userData }) => {
     });
 };
 
-module.exports = seed;
+module.exports = {seed};
```
Comment (Augs0) [inline db/seeds/seed.js]: There is no need to change this

File: API/app.js [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+    res.status(400).send({ msg: err.msg });
+  } else if (err.msg === "invalid order option") {
+    res.status(400).send({ msg: err.msg });
+  } else {
+    next(err);
+  }
```
Comment (Augs0) [inline API/app.js]: Instead of this error handler, at this stage I would add one for 500 errors as it's most helpful in dealing with psql errors

**Review by Augs0 — CHANGES_REQUESTED** [review]
Good job so far, Fame, just a few things to fix before moving on

File: API/app.js (hunk 1 above)
Comment (FameTii) [reply API/app.js]: Hi August, i keep getting app is not defined if i delete these lines. Am I misunderstanding the question?

File: API/app.test.js [hunk 8]
```diff
@@ -0,0 +1,32 @@
+const request = require("supertest");
+const seed = require("../db/seeds/seed");
+const db = require("../db/connection");
+const testData = require("../db/data/test-data/index");
+const app = require("./app");
+const { expect } = require("@jest/globals");
```
Comment (Augs0) [inline API/app.test.js:6]: Be sure to delete this or it will cause you problems. It's an auto import by vs code :)

**Review by Augs0 — CHANGES_REQUESTED** [review]
Please just remove the one thing mentioned here then you can move on. No need to send another PR :)

**Pull request merged**